src/pages/events.tsx

```diff
@@ -0,0 +1,404 @@
+                                                            </div>
+                                                        </form>
+                                                    </div>
+                                                )}
+                                                <button className="button-event" onClick={() => handleCancelClick(event.name + event.time)}>Cancel</button>
+                                            </td>
+                                        </tr>
+                                    </tbody>
+                                </table>
+                            </div>  
+                        ))}
+
+                        <p className="blue-border">Upcoming Events</p>
+
+                        {upcomingEvents.map((event) => (
+                            <div>
+                                <table className={getClassName(event.type)}>
+                                    <tbody>
+                                        <tr>
+                                            <td className="event-td left-td">{formatDateMonthDay(event.date)}<br></br>{listTypes(event.type)}</td>
+                                            <td className="event-td"> 
+                                                <Disclosure>
+                                                    {({open}) =>
+                                                        <>
+                                                            <div className="header-with-button">
+                                                                <Disclosure.Button className="disclosure-toggle">
+                                                                    {open ? (
+                                                                        <Image className="x" src={xMark} alt="close" />
+                                                                    ) : (
+                                                                        <Image className="arrow-down" src={leftChevron} alt="open" />
+                                                                    )}
+                                                                </Disclosure.Button><br></br>
+                                                                <div className="header-content">
+                                                                    <Image className="object-cover" src={logo} alt="gwg"/> {event.name}<br></br>
+                                                                    <span className="small-font">Time: {event.time}<br></br>{event.description}<br></br>Slots open: {event.slotsOpen} out of {event.totalSlots}</span><br></br>
+                                                                </div>
+                                                            </div>
+                                                            <Disclosure.Panel >
+                                                                {!registrationStates[event.name + event.time] && (
```

Comment:
When you move this to another component, there should be a state for open/close.